inc/api/api.class.php

```diff
@@ -2798,6 +2644,187 @@ protected function getFriendlyNames(
       return $_names;
    }
 
+   /**
+    * Get network ports
+    *
+    * @since 9.5
+    *
+    * @param array  $data
+    * @param array  $params         API parameters
+    * @param string $self_itemtype  Itemtype the API was called on
+    *
+    * @return array
+    */
+   protected function getNetworkPorts(
+      int $id,
+      string $itemtype
+   ) {
+      global $DB;
+
+      $_networkports = [];
+
+      if (!NetworkEquipment::canView()) {
+         $_networkports = $this->arrayRightError();
+      } else {
+         foreach (NetworkPort::getNetworkPortInstantiations() as $networkport_type) {
```

Comment:
Probably better to store `getNetworkPortInstantiations` result in a variable used for iteration

Reply:
Is it a bad practice to use a function result as a loop ? 
It feels more readable that way.

(I didn't write this, just moved the code).

Reply:
Doing this can lower performances; depending on what the called method is really doing (I'm not sure why, but this will kind of call the method at each iteration).

Reply:
Maybe you are thinking of this ? 

![image](https://user-images.githubusercontent.com/42734840/108969763-154b1400-7682-11eb-85a2-0348e12acfa2.png)

I've changed it anyway.